sequencer/src/hotshot_commitment.rs

```diff
@@ -83,10 +83,15 @@ pub async fn run_hotshot_commitment_task(opt: &CommitmentTaskOptions) {
     sequence(max, hotshot, contract).await;
 }
 
-async fn sequence(max_blocks: u64, hotshot: HotShotClient, contract: HotShot<Signer>) {
+async fn sequence(mut max_blocks: u64, hotshot: HotShotClient, contract: HotShot<Signer>) {
     loop {
         if let Err(err) = sync_with_l1(max_blocks, &hotshot, &contract).await {
             tracing::error!("error synchronizing with HotShot contract: {err}");
+            // If we are erroring because of a block gas limit exception, we can try again
+            // with fewer blocks next time
+            if err.to_string().contains("exceeds block gas limit") && max_blocks > 1 {
```

Comment:
This seems very fragile. What if something in the L1 or RPC or or ethers changes the wording slightly? But I guess there's not really a good way to extract the error cause from the error. What do you think about this instead:

If there is any error sending the transaction to L1, and the number of leaves being sent is `max_blocks`, assume we hit a gas error and decrease max blocks. `sync_with_l1` would have to be changed to return something like

```rust
enum SyncError {
  TransactionFailed { err: anyhow::Error, num_leaves: usize },
  Other(anyhow::Error),
}
```

but this would let us reliably detect this case without parsing error strings.

Reply:
This makes sense, `max_leaves` is the only parameter here we can really control here so seems reasonable for us to reduce if we max out and error